bin/test-example

```diff
@@ -0,0 +1,69 @@
+#!/bin/sh
+
+if [ $# -eq 0 ]; then
+    echo "usage: $0 /path/to/exercise/example"
+    exit 1
+fi
+
+exampledir=$(cd "$1" && pwd)
+
+examplename=$(basename "$exampledir")
+exercisedir=$(cd "$exampledir/../.." && pwd)
+exercisename=$(basename "$exercisedir")
+xhaskell="$(cd "$(dirname "$0")/.." && pwd)"
+
+buildfolder=$(mktemp -d)
```

Comment:
I met these problems last time I played with `.travis.yml`. I really wanted to use a temporary directory too...

If I remember correctly, `.stack-work` depends on the path and timestamps, but I never really tested the timestamp thing.